Cannot compile egui_demo_app on Linux

<!--
First look if there is already a similar bug report. If there is, upvote the issue with 👍

Please also check if the bug is still present in latest main! Do so by adding the following lines to your Cargo.toml:


[patch.crates-io]
egui = { git = "https://github.com/emilk/egui", branch = "main" }
# if you're using eframe:
eframe = { git = "https://github.com/emilk/egui", branch = "main" }
-->

**Describe the bug**
error: The platform you're compiling for is not supported by winit
error: could not compile `winit` (lib) due to 63 previous errors


**To Reproduce**
Steps to reproduce the behavior:
1. `cargo run --release -p egui_demo_app`

**Expected behavior**
demo app runs

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Browser: N/A
 - Version db87c712, also reproducible on 0.34.1
 - Window Manager: niri 25.11 (b35bcae)
 - Compiler: rustc 1.94.1 (e408947bf 2026-03-25)
 - cargo 1.94.1 (29ea6fb6a 2026-03-24)

**Logs**
```
error: The platform you're compiling for is not supported by winit
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/mod.rs:78:1
   |
78 | compile_error!("The platform you're compiling for is not supported by winit");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved import `self::platform`
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/platform_impl/mod.rs:34:15
   |
34 | pub use self::platform::*;
   |               ^^^^^^^^ could not find `platform` in `self`

error[E0432]: unresolved imports `crate::platform_impl::PlatformCustomCursor`, `crate::platform_impl::PlatformCustomCursorSource`
 --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/cursor.rs:8:28
  |
8 | use crate::platform_impl::{PlatformCustomCursor, PlatformCustomCursorSource};
  |                            ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `PlatformCustomCursorSource` in `platform_impl`
  |                            |
  |                            no `PlatformCustomCursor` in `platform_impl`

error[E0432]: unresolved import `crate::platform_impl::PlatformIcon`
 --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/icon.rs:1:5
  |
1 | use crate::platform_impl::PlatformIcon;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `PlatformIcon` in `platform_impl`

error[E0432]: unresolved import `crate::platform_impl::PlatformSpecificWindowAttributes`
 --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:7:34
  |
7 | use crate::platform_impl::{self, PlatformSpecificWindowAttributes};
  |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `PlatformSpecificWindowAttributes` in `platform_impl`

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1601:41
     |
1601 |         self.window.maybe_wait_on_main(|w| {
     |                                         ^
1602 |             w.available_monitors().into_iter().map(|inner| MonitorHandle { inner })
     |             - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1601 |         self.window.maybe_wait_on_main(|w: /* Type */| {
     |                                          ++++++++++++

error[E0282]: type annotations needed
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:50:41
   |
50 |         self.window.maybe_wait_on_main(|w| {
   |                                         ^
...
55 |             if let Some(Fullscreen::Exclusive(_)) = w.fullscreen().map(|f| f.into()) {
   |                                                     - type must be known at this point
   |
help: consider giving this closure parameter an explicit type
   |
50 |         self.window.maybe_wait_on_main(|w: /* Type */| {
   |                                          ++++++++++++

error[E0282]: type annotations needed
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:55:73
   |
55 |             if let Some(Fullscreen::Exclusive(_)) = w.fullscreen().map(|f| f.into()) {
   |                                                                         ^  - type must be known at this point
   |
help: consider giving this closure parameter an explicit type
   |
55 |             if let Some(Fullscreen::Exclusive(_)) = w.fullscreen().map(|f: /* Type */| f.into()) {
   |                                                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:498:41
    |
498 |         self.window.maybe_wait_on_main(|w| WindowId(w.id()))
    |                                         ^           - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
498 |         self.window.maybe_wait_on_main(|w: /* Type */| WindowId(w.id()))
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:566:41
    |
566 |         self.window.maybe_wait_on_main(|w| w.scale_factor())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
566 |         self.window.maybe_wait_on_main(|w: /* Type */| w.scale_factor())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:600:42
    |
600 |         self.window.maybe_queue_on_main(|w| w.request_redraw())
    |                                          ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
600 |         self.window.maybe_queue_on_main(|w: /* Type */| w.request_redraw())
    |                                           ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:639:42
    |
639 |         self.window.maybe_queue_on_main(|w| w.pre_present_notify());
    |                                          ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
639 |         self.window.maybe_queue_on_main(|w: /* Type */| w.pre_present_notify());
    |                                           ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:657:42
    |
657 |         self.window.maybe_queue_on_main(|w| w.reset_dead_keys())
    |                                          ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
657 |         self.window.maybe_queue_on_main(|w: /* Type */| w.reset_dead_keys())
    |                                           ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:681:41
    |
681 |         self.window.maybe_wait_on_main(|w| w.inner_position())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
681 |         self.window.maybe_wait_on_main(|w: /* Type */| w.inner_position())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:704:41
    |
704 |         self.window.maybe_wait_on_main(|w| w.outer_position())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
704 |         self.window.maybe_wait_on_main(|w: /* Type */| w.outer_position())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:742:47
    |
742 |         self.window.maybe_queue_on_main(move |w| w.set_outer_position(position))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
742 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_outer_position(position))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:761:41
    |
761 |         self.window.maybe_wait_on_main(|w| w.inner_size())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
761 |         self.window.maybe_wait_on_main(|w: /* Type */| w.inner_size())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:807:41
    |
807 |         self.window.maybe_wait_on_main(|w| w.request_inner_size(size))
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
807 |         self.window.maybe_wait_on_main(|w: /* Type */| w.request_inner_size(size))
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:824:41
    |
824 |         self.window.maybe_wait_on_main(|w| w.outer_size())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
824 |         self.window.maybe_wait_on_main(|w: /* Type */| w.outer_size())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:852:47
    |
852 |         self.window.maybe_queue_on_main(move |w| w.set_min_inner_size(min_size))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
852 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_min_inner_size(min_size))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:880:47
    |
880 |         self.window.maybe_queue_on_main(move |w| w.set_max_inner_size(max_size))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
880 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_max_inner_size(max_size))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:891:41
    |
891 |         self.window.maybe_wait_on_main(|w| w.resize_increments())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
891 |         self.window.maybe_wait_on_main(|w: /* Type */| w.resize_increments())
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:912:47
    |
912 |         self.window.maybe_queue_on_main(move |w| w.set_resize_increments(increments))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
912 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_resize_increments(increments))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:926:41
    |
926 |         self.window.maybe_wait_on_main(|w| w.set_title(title))
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
926 |         self.window.maybe_wait_on_main(|w: /* Type */| w.set_title(title))
    |                                          ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:947:47
    |
947 |         self.window.maybe_queue_on_main(move |w| w.set_transparent(transparent))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
947 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_transparent(transparent))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:961:47
    |
961 |         self.window.maybe_queue_on_main(move |w| w.set_blur(blur))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
961 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_blur(blur))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:975:47
    |
975 |         self.window.maybe_queue_on_main(move |w| w.set_visible(visible))
    |                                               ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
975 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_visible(visible))
    |                                                ++++++++++++

error[E0282]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:990:41
    |
990 |         self.window.maybe_wait_on_main(|w| w.is_visible())
    |                                         ^  - type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
990 |         self.window.maybe_wait_on_main(|w: /* Type */| w.is_visible())
    |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1011:47
     |
1011 |         self.window.maybe_queue_on_main(move |w| w.set_resizable(resizable))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1011 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_resizable(resizable))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1023:41
     |
1023 |         self.window.maybe_wait_on_main(|w| w.is_resizable())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1023 |         self.window.maybe_wait_on_main(|w: /* Type */| w.is_resizable())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1038:47
     |
1038 |         self.window.maybe_queue_on_main(move |w| w.set_enabled_buttons(buttons))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1038 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_enabled_buttons(buttons))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1049:41
     |
1049 |         self.window.maybe_wait_on_main(|w| w.enabled_buttons())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1049 |         self.window.maybe_wait_on_main(|w: /* Type */| w.enabled_buttons())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1061:47
     |
1061 |         self.window.maybe_queue_on_main(move |w| w.set_minimized(minimized))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1061 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_minimized(minimized))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1079:41
     |
1079 |         self.window.maybe_wait_on_main(|w| w.is_minimized())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1079 |         self.window.maybe_wait_on_main(|w: /* Type */| w.is_minimized())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1090:47
     |
1090 |         self.window.maybe_queue_on_main(move |w| w.set_maximized(maximized))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1090 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_maximized(maximized))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1101:41
     |
1101 |         self.window.maybe_wait_on_main(|w| w.is_maximized())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1101 |         self.window.maybe_wait_on_main(|w: /* Type */| w.is_maximized())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1133:47
     |
1133 | ..._queue_on_main(move |w| w.set_fullscreen(fullscreen.map(|f| f.into())))
     |                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1133 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_fullscreen(fullscreen.map(|f| f.into())))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1147:41
     |
1147 |         self.window.maybe_wait_on_main(|w| w.fullscreen().map(|f| f.into()))
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1147 |         self.window.maybe_wait_on_main(|w: /* Type */| w.fullscreen().map(|f| f.into()))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1147:64
     |
1147 |         self.window.maybe_wait_on_main(|w| w.fullscreen().map(|f| f.into()))
     |                                                                ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1147 |         self.window.maybe_wait_on_main(|w| w.fullscreen().map(|f: /* Type */| f.into()))
     |                                                                 ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1162:47
     |
1162 |         self.window.maybe_queue_on_main(move |w| w.set_decorations(decorations))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1162 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_decorations(decorations))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1176:41
     |
1176 |         self.window.maybe_wait_on_main(|w| w.is_decorated())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1176 |         self.window.maybe_wait_on_main(|w: /* Type */| w.is_decorated())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1190:47
     |
1190 |         self.window.maybe_queue_on_main(move |w| w.set_window_level(level))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1190 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_window_level(level))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1210:47
     |
1210 |         self.window.maybe_queue_on_main(move |w| w.set_window_icon(window_icon))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1210 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_window_icon(window_icon))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1257:47
     |
1257 | ..._queue_on_main(move |w| w.set_ime_cursor_area(position, size))
     |                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1257 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_ime_cursor_area(position, size))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1285:47
     |
1285 |         self.window.maybe_queue_on_main(move |w| w.set_ime_allowed(allowed))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1285 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_ime_allowed(allowed))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1300:47
     |
1300 |         self.window.maybe_queue_on_main(move |w| w.set_ime_purpose(purpose))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1300 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_ime_purpose(purpose))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1316:42
     |
1316 |         self.window.maybe_queue_on_main(|w| w.focus_window())
     |                                          ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1316 |         self.window.maybe_queue_on_main(|w: /* Type */| w.focus_window())
     |                                           ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1327:41
     |
1327 |         self.window.maybe_wait_on_main(|w| w.has_focus())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1327 |         self.window.maybe_wait_on_main(|w: /* Type */| w.has_focus())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1350:47
     |
1350 | ..._queue_on_main(move |w| w.request_user_attention(request_type))
     |                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1350 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.request_user_attention(request_type))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1371:47
     |
1371 |         self.window.maybe_queue_on_main(move |w| w.set_theme(theme))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1371 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_theme(theme))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1385:41
     |
1385 |         self.window.maybe_wait_on_main(|w| w.theme())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1385 |         self.window.maybe_wait_on_main(|w: /* Type */| w.theme())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1400:47
     |
1400 |         self.window.maybe_queue_on_main(move |w| w.set_content_protected(protected))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1400 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_content_protected(protected))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1411:41
     |
1411 |         self.window.maybe_wait_on_main(|w| w.title())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1411 |         self.window.maybe_wait_on_main(|w: /* Type */| w.title())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1428:47
     |
1428 |         self.window.maybe_queue_on_main(move |w| w.set_cursor(cursor))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1428 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_cursor(cursor))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1464:41
     |
1464 |         self.window.maybe_wait_on_main(|w| w.set_cursor_position(position))
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1464 |         self.window.maybe_wait_on_main(|w: /* Type */| w.set_cursor_position(position))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1489:41
     |
1489 |         self.window.maybe_wait_on_main(|w| w.set_cursor_grab(mode))
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1489 |         self.window.maybe_wait_on_main(|w: /* Type */| w.set_cursor_grab(mode))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1507:47
     |
1507 |         self.window.maybe_queue_on_main(move |w| w.set_cursor_visible(visible))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1507 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.set_cursor_visible(visible))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1524:41
     |
1524 |         self.window.maybe_wait_on_main(|w| w.drag_window())
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1524 |         self.window.maybe_wait_on_main(|w: /* Type */| w.drag_window())
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1543:41
     |
1543 |         self.window.maybe_wait_on_main(|w| w.drag_resize_window(direction))
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1543 |         self.window.maybe_wait_on_main(|w: /* Type */| w.drag_resize_window(direction))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1562:47
     |
1562 |         self.window.maybe_queue_on_main(move |w| w.show_window_menu(position))
     |                                               ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1562 |         self.window.maybe_queue_on_main(move |w: /* Type */| w.show_window_menu(position))
     |                                                ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1577:41
     |
1577 |         self.window.maybe_wait_on_main(|w| w.set_cursor_hittest(hittest))
     |                                         ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1577 |         self.window.maybe_wait_on_main(|w: /* Type */| w.set_cursor_hittest(hittest))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1589:41
     |
1589 | ...   self.window.maybe_wait_on_main(|w| w.current_monitor().map(|inner| MonitorHan...
     |                                       ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1589 |         self.window.maybe_wait_on_main(|w: /* Type */| w.current_monitor().map(|inner| MonitorHandle { inner }))
     |                                          ++++++++++++

error[E0282]: type annotations needed
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/winit-0.30.13/src/window.rs:1620:41
     |
1620 | ...   self.window.maybe_wait_on_main(|w| w.primary_monitor().map(|inner| MonitorHan...
     |                                       ^  - type must be known at this point
     |
help: consider giving this closure parameter an explicit type
     |
1620 |         self.window.maybe_wait_on_main(|w: /* Type */| w.primary_monitor().map(|inner| MonitorHandle { inner }))
     |                                          ++++++++++++

Some errors have detailed explanations: E0282, E0432.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `winit` (lib) due to 63 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/crates/egui_demo_app/Cargo.toml
+++ b/crates/egui_demo_app/Cargo.toml
@@ -23,7 +23,7 @@ crate-type = ["cdylib", "rlib"]
 
 
 [features]
-default = ["wgpu", "persistence"]
+default = ["wgpu", "persistence", "wayland", "x11"]
 
 web_app = ["http", "persistence"]
 

